### PR TITLE
Factor out the bootloader settings.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: Bootloader/firmware/Bootloader.X/nbproject/configurations.xml [hunk 1]
```diff
@@ -584,6 +500,7 @@
         <linkerTool>
           <linkerLibItems>
             <linkerLibFileItem>/opt/microchip/harmony/v1_02/bin/framework/peripheral/PIC32MX675F512H_peripherals.a</linkerLibFileItem>
+            <linkerLibFileItem>../../../../../../../opt/microchip/harmony/v1_02/bin/framework/peripheral/PIC32MX675F512H_peripherals.a</linkerLibFileItem>
```
Comment (peternewman) [inline Bootloader/firmware/Bootloader.X/nbproject/configurations.xml]: Make this absolute.

File: Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * ethrnet_sk2/bootloader_settings.c
```
Comment (peternewman) [inline Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h]: Correct this

File: Bootloader/firmware/src/system_config/number8/bootloader_settings.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * number8/bootloader_settings.c
```
Comment (peternewman) [inline Bootloader/firmware/src/system_config/number8/bootloader_settings.h]: Again

**Comment by peternewman** [comment]
Fix Travis. Generally LGTM

File: Bootloader/firmware/Bootloader.X/nbproject/configurations.xml (hunk 1 above)
Comment (nomis52) [inline Bootloader/firmware/Bootloader.X/nbproject/configurations.xml]: I think I'm losing the battle against this.

File: Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h (hunk 2 above)
Comment (nomis52) [inline Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h]: Done

File: Bootloader/firmware/Bootloader.X/nbproject/configurations.xml (hunk 1 above)
Comment (peternewman) [inline Bootloader/firmware/Bootloader.X/nbproject/configurations.xml]: Perhaps a pre-commit sed run?

File: Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * ethernet_sk2/bootloader_settings.c
```
Comment (peternewman) [inline Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h]: This is a .h, not .c, also adding the folder doesn't match what we have elsewhere, but I assume is intentional?

File: tests/system_config/bootloader_settings.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * ethrnet_sk2/bootloader_settings.c
```
Comment (peternewman) [inline tests/system_config/bootloader_settings.h]: SPaG again, and also .h not .c.

File: Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h (hunk 4 above)
Comment (nomis52) [inline Bootloader/firmware/src/system_config/ethernet_sk2/bootloader_settings.h]: Fixed. Yeah I added the path so it's clear what board this is for.

File: tests/system_config/bootloader_settings.h (hunk 5 above)
Comment (nomis52) [inline tests/system_config/bootloader_settings.h]: Fixed in another PR

**Pull request merged**